client/repo.c

```diff
@@ -537,6 +546,14 @@ TDNFGetRepoMD(
         BAIL_ON_TDNF_ERROR(dwError);
         nNeedDownload = 1;
     }
+    if(metalink)
+    {
+        dwError = TDNFAllocateStringPrintf(&pszRepoMetalink,
```

Comment:
TDNF_SAFE_FREE_MEMORY(pszRepoMetalink) is missed in cleanup.

Reply:
Yeah, Thanks!